Broadcast-Ext/OpenTok Live/SampleHandler.m

```diff
@@ -0,0 +1,215 @@
+    
+    // User has requested to start the broadcast. Setup info from the UI extension can be supplied but optional.
+    
+    // Provide session id and token
+    _broadcastHelper = [[OTBroadcastExtHelper alloc] initWithPartnerId:@""
+                                                             sessionId: @""
+                                                              andToken:@""
+                                                         videoCapturer:self];
+    _capture_queue = dispatch_queue_create("com.tokbox.OTBroadcastVideoCapture",
+                                           DISPATCH_QUEUE_SERIAL);
+    
+    _num_frames = 0;
+    [self destroyPixelBuffers];
+    [_broadcastHelper connect];
+}
+
+- (void)broadcastPaused {
+    // User has requested to pause the broadcast. Samples will stop being delivered.
+}
+
+- (void)broadcastResumed {
+    // User has requested to resume the broadcast. Samples delivery will resume.
+}
+
+- (void)broadcastFinished {
+    // User has requested to finish the broadcast.
+    [_broadcastHelper disconnect];
+}
+
+- (void) processPixelBuffer:(CVPixelBufferRef)pixelBuffer timeStamp:(CMTime)ts
+{
+    CIImage *ciImage = [CIImage imageWithCVPixelBuffer:pixelBuffer
+                                               options:nil];
+    //CGFloat imageWidth = CVPixelBufferGetWidth(pixelBuffer);
+    //CGFloat imageHeight = CVPixelBufferGetHeight(pixelBuffer);
+    CGFloat aspectRatio = 1;//imageHeight / imageWidth;
+    
+    ciImage = [self scaleFilterImage:ciImage
+                     withAspectRatio:aspectRatio scale:0.60];
+    
+    if(_pixelBufferPool == nil ||
+       CVPixelBufferGetWidth(pixelBuffer) != CVPixelBufferGetWidth(_pixelBuffer) ||
+       CVPixelBufferGetHeight(pixelBuffer) != CVPixelBufferGetHeight(_pixelBuffer))
+    {
+        [self destroyPixelBuffers];
+        [self createPixelBufferPoolWithWidth:ciImage.extent.size.width
+                                      height:ciImage.extent.size.height];
+        CVPixelBufferPoolCreatePixelBuffer(NULL, _pixelBufferPool, &_pixelBuffer);
+    }
+    
+    [_ciContext render:ciImage toCVPixelBuffer:_pixelBuffer];
+    
+    [self.videoCaptureConsumer consumeImageBuffer:_pixelBuffer
+                                      orientation:OTVideoOrientationUp
+                                        timestamp:ts
+                                         metadata:nil];
+}
+
+- (void)processSampleBuffer:(CMSampleBufferRef)sampleBuffer withType:(RPSampleBufferType)sampleBufferType {
+    
+    switch (sampleBufferType) {
+        case RPSampleBufferTypeVideo:
+        {
+            // Handle video sample buffer
+            
+            // Codec limits to avoid reaching over 50MB memory
+            // Tested on iPhone 8
+            // VP8  : Height = 800, Width = 450, fps = 10
+            // H264 : Height = 1068, Width = 600, fps = 15 or more
+            _num_frames++;
+            if (_num_frames % 3 != 0)
```

Comment:
I think this is a little bit confusing, I understand that it drops a frame each 3 frames, but it looks like a hack.

Reply:
There is no way to control the fps of incoming screen sharing video stream and we can't use  queue (since the memory limit < 50M). IMO The dropping frames is the simplest solution.

Reply:
Yeah, I think frameskipping is simplest solution and it works, but we should wrap this behaviour in a function something like ```if (should_skip_frame()) { ... }``` So it keeps clear to the reader of the sample